src/Spago/Repl.purs

```diff
@@ -6,21 +6,21 @@ import Spago.Prelude
 
 import Data.Map as Map
 import Registry.PackageName as PackageName
+import Spago.Config (BuildType(..), Package(..), PackageMap, PackageSet)
 import Spago.Registry (RegistryEnv)
-import Spago.Config (Package(..), PackageMap, PackageSet(..))
+import Spago.Registry as Registry
 
 -- TODO I guess this should be configurable
 supportPackageName :: PackageName
 supportPackageName = unsafeFromRight $ PackageName.parse "psci-support"
 
-supportPackage :: forall a. PackageSet -> Spago (RegistryEnv a) PackageMap
+supportPackage :: PackageSet -> Spago (RegistryEnv _) PackageMap
 supportPackage packageSet = do
-  { getMetadata, logOptions } <- ask
-  case packageSet of
-    PackageSet packages -> pure $ Map.filterWithKey (\k _v -> k == supportPackageName) packages
+  case packageSet.buildType of
+    PackageSetBuild _info packages -> pure $ Map.filterWithKey (\k _v -> k == supportPackageName) packages
     -- TODO: we should look in the "other" packages first
-    Registry _other -> do
-      maybeMetadata <- runSpago { logOptions } (getMetadata supportPackageName)
+    RegistrySolverBuild _other -> do
```

Comment:
A quick thought on solver vs. package set builds: should `spago init` require you to specify 'app' or 'lib' as an init type such that libraries use ranges off the bat and don't use a package set?

Reply:
> should spago init require you to specify 'app' or 'lib' as an init type such that libraries use ranges off the bat and don't use a package set?

We force users that publish their package to specify ranges, so I don't see a reason to enforce this upfront.

Reply:
Oh, if you mean "allow users to init a project without a package set", then we already have it: `spago init --use-solver`

Reply:
The memory of trying to get a Bower project to work and stepping through whole pages or solver errors in bright red ink is still fresh in me - sure, our solver is much more sensible than Bower's, but I feel the main reason the majority of the community is using package sets today is that they are just less complex to deal with and users can concentrate on actually writing the thing instead of dealing with builds. (as a user I would like that to be my experience at least, and I'm even in that small slice of the demographic that _likes_ dealing with builds!)

So yeah, I would prefer to keep package sets as default because they are easy, and have solver builds as the "yes yes I know what I'm about to get into" opt-in option.